### PR TITLE
Async Transact dialog for gas determination.

### PR DESCRIPTION
**Pull request by gavofyork**

### CODE REVIEW

File: alethzero/Transact.cpp [hunk 1]
```diff
@@ -64,6 +64,8 @@ Transact::Transact(MainFace* _c, QWidget* _parent):
 	setValueUnits(ui->valueUnits, ui->value, 0);
 
 	on_destination_currentTextChanged(QString());
+
+	startTimer(100);
```
Comment (arkpar) [inline alethzero/Transact.cpp]: m_gasCalcTimer supposed to hold this timer id?

File: alethzero/Transact.cpp [hunk 2]
```diff
@@ -356,23 +446,19 @@ GasRequirements Transact::determineGasRequirements()
...
 		{
-			lastGood = er;
-			if (haveUpperBound)
-				upperBound = mid;
-			else
-				haveUpperBound = true;
+			m_lastGood = lastGood;
+			m_lowerBound = m_startLowerBound;
+			m_upperBound = m_startUpperBound;
+			startTimer(0);
```
Comment (arkpar) [inline alethzero/Transact.cpp]: Can this be called while the previous timer is still active? Also startTimer in contructor uses different interval

File: alethzero/Transact.cpp (hunk 1 above)
Comment (gavofyork) [inline alethzero/Transact.cpp]: yes - fixed in latest.

File: alethzero/Transact.cpp (hunk 2 above)
Comment (gavofyork) [inline alethzero/Transact.cpp]: yes - both points addressed in recent fix.

**Pull request merged**